### ISSUE TITLE
False positive (expression-not-assigned) in list comprehension

### ISSUE THREAD

**Issue by BennyTheSen** [PyCQA/pylint#3397]
### Steps to reproduce
Example:
```python
""" docstring """
T_LIST = ["a", "b", "test_abcd", "c"]
[T_LIST.remove(line) for line in T_LIST if "test" in line]
T_LIST.remove("a")

print(T_LIST)

```

### Current behavior
```bash
$ pylint example.py 
************* Module example
example.py:3:0: W0106: Expression "[T_LIST.remove(line) for line in T_LIST if 'test' in line]" is assigned to nothing (expression-not-assigned)

-------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 10.00/10, -2.50)

```

### Expected behavior
Pylint should not mark the list.remove() function as not assigned when used in list comprehension. 
There's no error if used outside of a list comprehension.

### pylint --version output
```bash
$ pylint --version
pylint 2.4.4
astroid 2.3.3
Python 3.7.3 (default, Jul  5 2019, 11:01:28) 
[GCC 5.4.0 20160609]


```

**Comment by PCManticore** [comment]
Please write a normal `for-loop`, we're not going to support this use case.
